clients.go

```diff
@@ -61,12 +67,20 @@ func newFastHTTPClient(opts *clientOpts) client {
 	c.headers = headersToFastHTTPHeaders(opts.headers)
 	c.url, c.method, c.body = opts.url, opts.method, opts.body
 	c.bodProd = opts.bodProd
+	c.l = opts.l
 	return client(c)
 }
 
 func (c *fasthttpClient) do() (
 	code int, msTaken uint64, err error,
 ) {
+	// Prepare the Request Parameters with lua script
+	if c.l != nil {
+		uri, _ := url.Parse(c.url)
+		uri.Path, c.method, *c.body, _ = executeLuaScript(c.l, uri.Path, c.method, *c.body)
```

Comment:
I ran into some panics, when i launched your version of the program. I suspect that these come from this line, because `c.l` is used by multiple goroutines simultaneously and `LState` is not goroutine-safe as pointed out in [this section](https://github.com/yuin/gopher-lua#goroutines) of GopherLua's README.